dbms/src/Encryption/FileProvider.cpp

```diff
@@ -32,7 +35,15 @@ RandomAccessFilePtr FileProvider::newRandomAccessFile(
     const ReadLimiterPtr & read_limiter,
     int flags) const
 {
-    RandomAccessFilePtr file = std::make_shared<PosixRandomAccessFile>(file_path_, flags, read_limiter);
+    RandomAccessFilePtr file;
+    if (S3::S3FilenameView::fromKey(file_path_).isValid())
+    {
+        file = S3::S3RandomAccessFile::create(file_path_);
+    }
+    else
+    {
+        file = std::make_shared<PosixRandomAccessFile>(file_path_, flags, read_limiter);
+    }
     auto encryption_info = key_manager->getFile(encryption_path_.full_path);
```

Comment:
This brings me a question about the feature compatibility between S3 storage and encryption.
If encryption is enabled, then I think we get into trouble... We can not find the encryption key (which only exists in 1 write node) to decrypt it when reading from the read node or another write node.

Reply:
in conclusion: encryption at rest can not be enabled when using S3 as the main storage so far. And we will list it as a limitation in doc.